unit/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -306,6 +308,46 @@ TEST_CASE(
             "B0", {}, smt_identifier_termt{"bar", smt_bit_vector_sortt{8}}}});
     }
   }
+    SECTION("Handle single nondet_symbol")
+    {
+        // Using the standard way to create nondet_symbol_exprt.
+        ::symex_nondet_generatort generator;
+        const nondet_symbol_exprt nondet_symbol0 = generator(bool_typet{}, source_locationt{});
+        const smt_identifier_termt nondet_symbol_term0{nondet_symbol0.get_identifier(),
+                                                       smt_bool_sortt{}};
+        test.sent_commands.clear();
+        test.procedure.handle(nondet_symbol0);
+        REQUIRE(
+                test.sent_commands ==
+                std::vector<smt_commandt>{
+                        smt_declare_function_commandt{nondet_symbol_term0, {}},
+                        smt_define_function_commandt{"B0", {}, nondet_symbol_term0}
+                }
+        );
+    }
+    SECTION("Handle multiple nested nondet_symbol")
+    {
+        ::symex_nondet_generatort generator;
+        const nondet_symbol_exprt nondet_symbol1 = generator(bitvector_typet{ID_bv, 42}, source_locationt{});
```

Comment:
:no_entry_sign: Please don't use `ID_bv` directly in the construction of the type. This constructor should only be used by the sub-classes, as there is no check in the constructor that the supplied ID corresponds to a valid sub class. Therefore I would prefer that you use `bv_typet{42}` instead of `bitvector_typet{ID_bv, 42}`.

Reply:
Right.
Changed